watertap/tools/oli_api/client.py

```diff
@@ -123,29 +146,28 @@ def _upload_dbs_file(self, dbs_file_path):
 
         try:
             with open(dbs_file_path, "rb") as file:
-                files = {"files": file}
-                req_result = self.request_auto_login(
-                    lambda headers: requests.post(
-                        self.credential_manager.upload_dbs_url,
-                        headers=headers,
-                        files=files,
-                    )
+                req_result = requests.post(
+                    self.credential_manager.upload_dbs_url,
+                    headers=self.credential_manager.headers,
+                    files={"files": file},
                 )
-            dbs_file_id = req_result["file"][0]["id"]
+            _logger.debug(f"DBS file result: {json.loads(req_result.text)}")
+            dbs_file_id = json.loads(req_result.text)["file"][0]["id"]
+            _logger.info(f"DBS file id is {dbs_file_id}")
             self.session_dbs_files.append(dbs_file_id)
             return dbs_file_id
         except:
-            raise IOError(" Failed to upload DBS file. Ensure specified file exists.")
+            raise RuntimeError(
+                " Failed to upload DBS file. Ensure specified file exists."
+            )
 
     def _create_dbs_dict(self, chemistry_source, phases, model_name):
-
         """
         Creates dict for chemistry-builder to later generate a DBS file id.
 
-        :param chemistry_source: OLI-compatible ion names as a Pyomo Set, list, or dict,
-            where the keys are ion names
-        :param model_name: Name for model as a string
-        :param phases: OLI-compatible phases
+        :param chemistry_source: path (str), dict, or state block containing chemistry info
```

Comment:
Here, `chemistry_source` can still be an iterable with oli names as keys. However, it can also be a string pointing to a local dbs file to upload. Once the PR is resolved I'll prioritize updating the doc strings for clarity/consistency